### ISSUE TITLE
🐛 Bug: Model eos1579:metabokiller fails with `EmptyOutputError`

### ISSUE THREAD

**Issue by sayantani11** [ersilia-os/ersilia#369]
### Describe the bug.

Previously for a same kind of error in a different model, it was suggested that maybe it is because of git-lfs being broken, but even after doing the necessary steps the same `EmptyOutputError` pops up. The same error even occurs in the Google Colab as well.

### Expected behavior.

👍🏼model eos1579 fetched successfully!

### Screenshots.

[eos1579.txt](https://github.com/ersilia-os/ersilia/files/9801811/eos1579.txt)

### Operating environment

WSL2 Ubuntu 18.04.5

**Comment by sayantani11** [comment]
@GemmaTuron I was thinking like, what about this problem:
```sh
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
bentoml 0.11.0 requires sqlalchemy<1.4.0,>=1.3.0, but you have sqlalchemy 1.4.36 which is incompatible.
```
How to tackle this?

**Comment by GemmaTuron** [comment]
Hi @sayantani11 ! 

If you check a few lines below you will see how this is dealt with directly (line 400)

  Attempting uninstall: sqlalchemy
    Found existing installation: SQLAlchemy 1.4.36
    Uninstalling SQLAlchemy-1.4.36:
      Successfully uninstalled SQLAlchemy-1.4.36
Successfully installed sqlalchemy-1.3.24

This is not the source of the error... I am attaching here a log I generated in my linux machine, and you will see the model is fetched successfull (in line 402 is reinstalling the right version of sqlalchemy)
Maybe you can have a look at what is showing different? (line 400 vs 402, so some output is differing before that) and it will help us find the source of the problem!
[eos1579.log](https://github.com/ersilia-os/ersilia/files/9802513/eos1579.log)

thanks :)

**Comment by GemmaTuron** [comment]
I am also linking here issue #338 from @femme-js as it is related to the same model
I can successfully fetch it from my CLI, so there must be some different installation setup.
Can both of you clarify:
- What system are you using
- What conda version
- What pip version
- What python version
- What slqalchemy we have on the ersilia environment (if it is installed) and in the eos1579 environment

**Comment by GemmaTuron** [comment]
Also mark this on excel as ongoing discussion but don't get stuck here and proceed with other tasks meanwhile

**Comment by sayantani11** [comment]
This is the colab output:
```sh
23:49:06 | DEBUG    | [{'input': {'key': 'LUHMMHZLDLBAKX-UHFFFAOYSA-N', 'input': 'CC1C2C(CC3(C=CC(=O)C(=C3C2OC1=O)C)C)O', 'text': 'CC1C2C(CC3(C=CC(=O)C(=C3C2OC1=O)C)C)O'}, 'output': None}, {'input': {'key': 'QRXWMOHMRWLFEY-UHFFFAOYSA-N', 'input': 'C1=CN=CC=C1C(=O)NN', 'text': 'C1=CN=CC=C1C(=O)NN'}, 'output': None}]
23:49:06 | ERROR    | Ersilia exception class:
EmptyOutputError

Detailed error:
Model API eos1579:predict did not produce an output

Hints:
- Visit the fetch troubleshooting site

🚨🚨🚨 Something went wrong with Ersilia 🚨🚨🚨

Error message:

Ersilia exception class:
EmptyOutputError

Detailed error:
Model API eos1579:predict did not produce an output
```

**Comment by GemmaTuron** [comment]
> This is the colab output:
> 
> ```shell
> 23:49:06 | DEBUG    | [{'input': {'key': 'LUHMMHZLDLBAKX-UHFFFAOYSA-N', 'input': 'CC1C2C(CC3(C=CC(=O)C(=C3C2OC1=O)C)C)O', 'text': 'CC1C2C(CC3(C=CC(=O)C(=C3C2OC1=O)C)C)O'}, 'output': None}, {'input': {'key': 'QRXWMOHMRWLFEY-UHFFFAOYSA-N', 'input': 'C1=CN=CC=C1C(=O)NN', 'text': 'C1=CN=CC=C1C(=O)NN'}, 'output': None}]
> 23:49:06 | ERROR    | Ersilia exception class:
> EmptyOutputError
> 
> Detailed error:
> Model API eos1579:predict did not produce an output
> 
> Hints:
> - Visit the fetch troubleshooting site
> 
> 🚨🚨🚨 Something went wrong with Ersilia 🚨🚨🚨
> 
> Error message:
> 
> Ersilia exception class:
> EmptyOutputError
> 
> Detailed error:
> Model API eos1579:predict did not produce an output
> ```

Thanks @sayantani11 
From the colab notebooks is best if you can share the link to them when they have run so I can see the whole error.

**Comment by GemmaTuron** [comment]
> I am also linking here issue #338 from @Femme-js as it is related to the same model I can successfully fetch it from my CLI, so there must be some different installation setup. Can both of you clarify:
> 
> * What system are you using
> * What conda version
> * What pip version
> * What python version
> * What slqalchemy we have on the ersilia environment (if it is installed) and in the eos1579 environment

Hi @sayantani11 and  @femme-js, re-posting these questions! @femme-js very helpful if you paste your current version, then do the updates as I suggested on #338 and try fetching again :)

**Comment by sayantani11** [comment]
@GemmaTuron 
System = WSL Ubuntu 18.04
Conda = conda 22.9.0
pip = pip 22.2.2
python = Python 3.7.13
slqalchemy = says not installed and eos1579 environment i don't see but this is there `eosbase-bentoml-0.11.0-py37`

**Comment by GemmaTuron** [comment]
Hi @sayantani11 :
The log of the error is showing the git-lfs issue, so for some reason it is not able to find it in your system even though it's installed.If you run
`conda env list`

It does not show eos1579?

**Comment by sayantani11** [comment]
> Hi @sayantani11 : The log of the error is showing the git-lfs issue, so for some reason it is not able to find it in your system even though it's installed.If you run `conda env list`
> 
> It does not show eos1579?

No, it doesn't show eos1579. I'll try the entire thing again let you know.

**Comment by GemmaTuron** [comment]
Let's give this one last try to make sure git lfs is installed in your WSL
From the WSL CLI:
- Update Conda
- Update Pip
- Make sure git is actually installed (sudo apt install git)
- Fetch git lfs and install again  (ubuntu: https://github.com/git-lfs/git-lfs/wiki/Installation#ubuntu)
- Restart system

**Comment by sayantani11** [comment]
> Let's give this one last try to make sure git lfs is installed in your WSL From the WSL CLI:
> 
> * Update Conda
> * Update Pip
> * Make sure git is actually installed (sudo apt install git)
> * Fetch git lfs and install again  (ubuntu: https://github.com/git-lfs/git-lfs/wiki/Installation#ubuntu)
> * Restart system

[eos1579.txt](https://github.com/ersilia-os/ersilia/files/9847633/eos1579.txt)
The fetch was unsuccessful but there's no git-lfs error in the logs like i can see

**Comment by sayantani11** [comment]
> Hi @sayantani11 : The log of the error is showing the git-lfs issue, so for some reason it is not able to find it in your system even though it's installed.If you run `conda env list`
> 
> It does not show eos1579?

and I got why this happened cause i did `ersilia delete eos1579`

**Comment by GemmaTuron** [comment]
Thanks  @sayantani11 !

Let's pause this here as we look into what might be causing the empty output error. Git LFS is now properly installed ;)

**Comment by pauline-banye** [comment]
Update on eos1579 @GemmaTuron @miquelduranfrigola. 

- Operating environment: WSL on Windows 10 
- Ubuntu version: 20.04
- Pip version: 22.3.1
- Python version: 3.7.15
- Anaconda version: 4.14.

Following the steps I took for fetching eos3ae7, I did not clone Isaura to avoid dependency issues with bentoml.

I fetched this model multiple times and encountered the same error consistently `Cannot choose from an empty sequence`. 

![1579 error](https://user-images.githubusercontent.com/63536707/207026689-d7ac5731-c794-4185-85a0-3bba759285df.png)

[eos1579.log](https://github.com/ersilia-os/ersilia/files/10207164/eos1579.log)

**Comment by GemmaTuron** [comment]
Update:
Mdeol eos1579 fail at fetch time both in CLI and Colab.
CLI error i not very informative`cannot choose from an empty sequence` --> [eos1579_cli.txt](https://github.com/ersilia-os/ersilia/files/10427073/eos1579_cli.txt)
Colab error: missing ipykernel package --> [eos1579_colab.txt](https://github.com/ersilia-os/ersilia/files/10427072/eos1579_colab.txt)

I also think this model might have some unnecessary functions inside and removing them would make it easier to work with (it is particularly heavy).
BTW ipykernel package is indeed not installed in conda env eos1579

**Comment by zerioi9** [comment]
Update on model eos1579 @GemmaTuron @pauline-banye 

Initially while fetching the model using Google Colab it gave the error `ModeuleNotFoundError: No module named 'ipykernel'`. I then installed ipykernel using the command  `!pip install ipykernel` unfortunately it did not work, it gave the same error as before. So after researching about it on the internet I found that ipykernel is supported on Python >=3.8 but the ready-to-go Google Colaboratory notebook uses Python 3.7. I am a bit confused now, how should I approach this further?
The log file--> [eos1579_fetch_colab.txt](https://github.com/ersilia-os/ersilia/files/10949918/eos1579_fetch_colab.txt)
